core/pkg/server/sender.go

```diff
@@ -675,6 +695,7 @@ func (s *Sender) sendRun(record *service.Record, run *service.RunRecord) {
 			nil,                              // summaryMetrics
 		)
 		if err != nil {
+			// s.cancel()
```

Comment:
```suggestion
			// s.cancel()
```

should we remove this?